### PR TITLE
Make prometheus-node-exporter executable

### PR DESCRIPTION
**Pull request by matthewtorr-msft**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

By specifying the name of a directory to build, the spec file was instructing go build to create an archive file, not an executable. Using a plain go build command makes it build an executable as intended.

###### Change Log  

- See above.
- Also added a sense test under %check that the output file can be run.

###### Does this affect the toolchain?  

No

###### Associated issues  

- OS Bug 38396447

###### Test Methodology

I installed my locally-built RPM on a Mariner VM. prometheus-node-exporter is now an executable file, which is an improvement already:

```
centos@ci1382011-mt5-s-sde-2 [ ~ ]$ file $(which node_exporter)
/usr/bin/node_exporter: symbolic link to prometheus-node-exporter
centos@ci1382011-mt5-s-sde-2 [ ~ ]$ file $(which prometheus-node-exporter)
/usr/bin/prometheus-node-exporter: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, Go BuildID=GjRwdH7oW4SHZmPKOYhv/Q8FwkpBJy2S4TQGzKns_/3vLm-CTNOqf6j_RroK_9/VmruVG3o_a88Nenl3dw9, not stripped
```

Additionally after starting the systemd service, it ran and did not crash, and curling http://127.0.0.1:9100/metrics returned some sensible-looking metrics.

### CODE REVIEW

File: SPECS/prometheus-node-exporter/prometheus-node-exporter.spec [hunk 1]
```diff
@@ -58,7 +58,11 @@ LDFLAGS="-X github.com/prometheus/common/version.Version=%{version}      \
          -X github.com/prometheus/common/version.Branch=tarball          \
          -X github.com/prometheus/common/version.BuildDate=%{build_date} \
          -X github.com/ncabatoff/process-exporter/version.GoVersion=%{go_version}"
-go build -ldflags "$LDFLAGS" -mod=vendor -v -a -tags "$BUILDTAGS" -o bin/node_exporter ./collector
+go build -ldflags "$LDFLAGS" -mod=vendor -v -a -tags "$BUILDTAGS" -o bin/node_exporter
+
+%check
+make test
+bin/node_exporter --help
```
Comment (mfrw) [inline SPECS/prometheus-node-exporter/prometheus-node-exporter.spec:65]: In my opinion running this command would override the results from the `make test`.
Even if the tests fail, our tooling wont be able to catch it, as it depends on the exit status of the last thing that runs in the check section.
IMO we should swap them:
```suggestion
bin/node_exporter --help
make test
```

Comment (matthewtorr-msft) [reply SPECS/prometheus-node-exporter/prometheus-node-exporter.spec:65]: Thank you for the suggestion!

It's kind of sad that only the last thing has its exit status checked. Would something like this be more appropriate to check both things?

```suggestion
bin/node_exporter --help && make test
```

Comment (mfrw) [reply SPECS/prometheus-node-exporter/prometheus-node-exporter.spec:65]: Yep, this looks good to me :)

Comment (matthewtorr-msft) [reply SPECS/prometheus-node-exporter/prometheus-node-exporter.spec]: Done :)

**Comment by matthewtorr-msft** [comment]
I deliberately haven't applied any markups from the spec linting job, because experience from previous PRs has shown that the suggestions it's asking for are likely to cause bugs.

File: SPECS/prometheus-node-exporter/prometheus-node-exporter.spec [hunk 2]
```diff
@@ -73,9 +76,6 @@ install -Dpm0644 example-rules.yml %{buildroot}%{_datadir}/prometheus/node-expor
 install -Dpm0644 %{SOURCE5} %{buildroot}%{_sysconfdir}/logrotate.d/%{name}
 mkdir -vp %{buildroot}%{_sharedstatedir}/prometheus/node-exporter
 
-%check
-make test
```
Comment (PawelWMS) [inline SPECS/prometheus-node-exporter/prometheus-node-exporter.spec:77]: Why are we removing the tests?

Comment (PawelWMS) [reply SPECS/prometheus-node-exporter/prometheus-node-exporter.spec:77]: Nevermind, I see you just moved them elsewhere. Not sure what the `bin/node_exporter --help` is for, though. And could we move the whole section where it was - we usually keep them after the `%install` section.

File: SPECS/prometheus-node-exporter/prometheus-node-exporter.spec [hunk 3]
```diff
@@ -107,6 +107,9 @@ getent passwd 'prometheus' >/dev/null || useradd -r -g 'prometheus' -d '%{_share
 %dir %attr(0755,prometheus,prometheus) %{_sharedstatedir}/prometheus/node-exporter
 
 %changelog
+* Tue Mar 29 2022 Matthew Torr <matthewtorr@microsoft.com> - 1.3.1-7
```
Comment (PawelWMS) [inline SPECS/prometheus-node-exporter/prometheus-node-exporter.spec:110]: Please also increment the `Release` tag value.

File: SPECS/prometheus-node-exporter/prometheus-node-exporter.spec (hunk 2 above)
Comment (matthewtorr-msft) [reply SPECS/prometheus-node-exporter/prometheus-node-exporter.spec:77]: The point is that "make test" did not catch the bug that this PR fixes, namely that the prometheus-node-exporter "binary" is not an executable file. Therefore I added a simple test that the binary can be run.

I can move the check section back if that's the house style.

File: SPECS/prometheus-node-exporter/prometheus-node-exporter.spec (hunk 3 above)
Comment (matthewtorr-msft) [reply SPECS/prometheus-node-exporter/prometheus-node-exporter.spec:110]: Good catch, thank you.

**Review by jslobodzian — APPROVED** [review]

**Review by jslobodzian — APPROVED** [review]

**Review by oliviacrain — APPROVED** [review]

**Pull request merged**